wallet/wallet.go

```diff
@@ -838,6 +838,148 @@ func (w *Wallet) syncToBirthday() (*waddrmgr.BlockStamp, error) {
 	return birthdayStamp, nil
 }
 
+// recovery attempts to recover any unspent outputs that pay to any of our
+// addresses. The birthdayStamp is used to indicate the starting point of our
+// recovery, as it's not possible for us to create any addresses before our
+// wallet's birthday.
+func (w *Wallet) recovery(birthdayStamp *waddrmgr.BlockStamp) error {
```

Comment:
At a glance a large degree of code duplication exists in this method and the one introduced in the prior commit.

Reply:
Fixed, thanks @halseth for the suggestion!